services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/steps/FlowCrudSteps.java

```diff
@@ -95,62 +97,42 @@
     public void defineFlowsOverActiveSwitches() {
-        FlowSetBuilder builder = new FlowSetBuilder();
-
-        final List<TopologyDefinition.Switch> switches = topologyDefinition.getActiveSwitches();
-        // check each combination of active switches for a path between them and create a flow definition if the path exists
-        switches.forEach(srcSwitch ->
-                switches.forEach(dstSwitch -> {
-                    // skip the same switch flow and reverse combination of switches
-                    if (srcSwitch.getDpId().compareTo(dstSwitch.getDpId()) >= 0) {
-                        return;
-                    }
-
-                    // test only bi-directional flow
-                    List<PathInfoData> forwardPath = topologyEngineService
-                            .getPaths(srcSwitch.getDpId(), dstSwitch.getDpId());
-                    List<PathInfoData> reversePath = topologyEngineService
-                            .getPaths(dstSwitch.getDpId(), srcSwitch.getDpId());
-                    if (!forwardPath.isEmpty() && !reversePath.isEmpty()) {
-                        String flowId = format("%s-%s", srcSwitch.getName(), dstSwitch.getName());
-                        builder.addFlow(flowId, srcSwitch, dstSwitch);
-                    }
-                })
-        );
-
-        flows = builder.getFlows();
+        flows = flowCalculator.allActiveSwitchesFlows();
     }
 
     @Given("^flows defined over active traffgens in the reference topology$")
     public void defineFlowsOverActiveTraffgens() {
-        FlowSetBuilder builder = new FlowSetBuilder();
-
-        final List<TopologyDefinition.TraffGen> traffGens = topologyDefinition.getActiveTraffGens();
-        // check each combination of active traffGens and create a flow definition
-        traffGens.forEach(srcTraffGen -> {
-            TopologyDefinition.Switch srcSwitch = srcTraffGen.getSwitchConnected();
-            traffGens.forEach(dstTraffGen -> {
-                TopologyDefinition.Switch dstSwitch = dstTraffGen.getSwitchConnected();
-                // skip the same switch flow and reverse combination of switches
-                if (srcSwitch.getDpId().compareTo(dstSwitch.getDpId()) >= 0) {
-                    return;
-                }
+        flows = flowCalculator.allActiveTraffgenFlows();
+    }
 
-                String flowId = format("%s-%s", srcSwitch.getName(), dstSwitch.getName());
-                builder.addFlow(flowId, srcSwitch, srcTraffGen.getSwitchPort(), dstSwitch, dstTraffGen.getSwitchPort());
-            });
-        });
+    @Given("Create (\\d+) flows? with A Switch used and at least (\\d+) alternate paths? between source and "
+            + "destination switch and (\\d+) bandwidth")
+    public void flowsWithAlternatePaths(int flowsAmount, int alternatePaths, int bw) {
+        topologyUnderTest.getFlowIsls().putAll(flowCalculator.createFlowsWithASwitch(flowsAmount, alternatePaths, bw));
```

Comment:
2 issues:
- It's better to split such invocation chains to improve readability. E.g. local variable.
- Bad practice to modify a collection obtained via getter.